packages/shared/src/lib/core/activity/utils/generateActivityFromPeristedTransaction.ts

```diff
@@ -0,0 +1,79 @@
+import { BlockscoutTransactionStatus, IBlockscoutTransaction } from '@auxiliary/blockscout/interfaces'
+import { IAccountState } from '@core/account/interfaces'
+import { getTransferInfoFromTransactionData } from '@core/layer-2/utils/getTransferInfoFromTransactionData'
+import { IChain } from '@core/network'
+import { PersistedTransaction } from '@core/transactions/stores'
+import { LegacyTxData, TransactionType } from '@ethereumjs/tx'
+import { TransactionReceipt } from 'web3-core'
+import { ActivityType } from '../enums'
+import { Activity, PersistedEvmTransaction } from '../types'
+import { generateNftActivity } from './evm/generateNftActivity'
+import { generateSmartContractActivity } from './evm/generateSmartContractActivity'
+import { generateTokenActivity } from './evm/generateTokenActivity'
+
+export async function generateActivityFromPersistedTransaction(
+    persistedTransaction: PersistedTransaction,
+    chain: IChain,
+    account: IAccountState
+): Promise<Activity | undefined> {
+    const transaction = persistedTransaction.blockscout
+        ? buildPersistedEvmTransactionFromBlockscoutTransaction(persistedTransaction.blockscout)
```

Comment:
rather than turn everything into the internal transaction type can we parse blockscout data to get useful information